src/components/WebChat/WebChat.js

```diff
@@ -1,80 +1,95 @@
-  }, [chatHistory]);
+    if(isChatBoxVisible)
+      chatBox.current.scrollTop = chatBox.current.scrollHeight;
+  }, [chatHistory, isChatBoxVisible]);
 
-  useEffect(
-    (event) => {
-      if (isChatBoxVisible)
-        chatBox.current.scrollTop = chatBox.current.scrollHeight;
-    },
-    [chatHistory, isChatBoxVisible]
-  );
+  const newMessage = () => {
+
+    let users = [
+      'a',
+      'b',
+    ];
+
+    let randomUser = Math.floor(Math.random()*2);
+
+    return {
+      id: Math.random().toString(32),
+      user: users[randomUser],
+      message: message,
+    }
 
+  }
+  
   const onClickToggleChatBox = () => setIsChatBoxVisible(!isChatBoxVisible);
 
-  const onClickAddMessage = () => {};
+  const onClickAddMessage = () => {
+    if(message.trim().length===0) return;
+    setChatHistory([...chatHistory, newMessage()]); 
+    setMessage('');
+  };
 
-  const onChangeMessage = (event) => {};
+  const onChangeMessage = (event) => {
+    setMessage(event.target.value);
+  };
+
+  const onKeyUpInput = (event) => {
+    const { keyCode } = event;
+    if(keyCode === 13 && event.ctrlKey) {
+      onClickAddMessage();
+    }
+  }
 
   return (
-    <div className="webchat-container">
-      <div
-        className={`webchat-panel-inner webchat-toggle--${
-          isChatBoxVisible ? "show" : "hide"
-        }`}
-      >
-        <div className="webchat-bar">
-          <p className="webchat-bar-spacing">How may we help you?</p>
+    <div className="lqe-wct">
+      <div className={`lqe-wct__main lqe-wct__main__toggle--${isChatBoxVisible ? "show" : "hide"}`}>
+        <div className="lqe-wct__main__hdr">
+          <p className="lqe-wct__main__hdr_text_s">How may I help you?
+          </p>
           <BsX onClick={onClickToggleChatBox}></BsX>
         </div>
-
-        <div id="webchat-window" ref={chatBox} className="webchat-window">
+        <div ref={chatBox} className="lqe-wct__main__message-list">
           {chatHistory.map((chat) => (
-            <div key={chat.id} className={`webchat-message-${chat.user}`}>
-              <div className="webchat-dot"></div>
-              <div className={`webchat-chatbubble-${chat.user}`}>
+            <div key={chat.id} className={`lqe-wct__main__message-list__message lqe-wct__main__message-list__message${chat.user}`}>
+              <div className="message__dot"></div>
+              <div className={`message__chatbubble message__chatbubble${chat.user}`}>
                 {chat.message}
               </div>
             </div>
           ))}
         </div>
-
-        <div className="webchat-message-submit">
-          <input
-            onChange={onChangeMessage}
-            className="message-input"
-            placeholder="Enter your message"
+        <div className="lqe-wct__message__input__wrap">
+          <input 
+            onChange={onChangeMessage} 
+            onKeyUp={onKeyUpInput} value={message}
+            className="lqe-wct__message__input lqe-wct__message__input_s" 
```

Comment:
Should these classes contain main within the name as they are children of the lqe-wct__main class?

Reply:
The thinking here was since they are inside an item inside a list they do not need to contain the full path of the parent.

This is why I only have 

message__dot
message__chatbubble message__chatbubblea

So by that logic I was trying to copy the example we saw on that BEM site

Was trying to follow something along this lines

<ul class="l-grid">
    <li class="l-grid__item">
        <div class="c-card">
            <div class="c-card__header">
                […]
            </div>
            <div class="c-card__body">
                […]
            </div>
        </div>
    </li>

By doing this

          <div className="lqe-wct__main__message-list">
            <div className="lqe-wct__main__message-list__message lqe-wct__main__message-list__messagea">
              <div className="message__dot" />
              <div className="message__chatbubble message__chatbubblea">
                this is user ** speaking this is user ** speaking this is user ** speaking this is user ** speaking 
              </div>
            </div>
          </div>